core/tests/test_utils.py

```diff
@@ -143,16 +148,43 @@ class TestBase(unittest.TestCase):
     }
 
     def _get_unicode_test_string(self, suffix):
+        """Returns a string for unicode compatibility test.
+
+        Args:
+            suffix: string. Appended to unicode string.
+
+        Returns:
+            string. Unicode test string.
+        """
         return '%s%s' % (self.UNICODE_TEST_STRING, suffix)
 
     def setUp(self):
+        """Sets up environment for Test Base
+            (implemented in inherited class).
+
+        Exception:
+            NotImplementedError. Raises exception if the method is not
+                implemented in inherited class.
+        """
         raise NotImplementedError
 
     def tearDown(self):
+        """Deactivates environment of Test Base
+            such as logout, delete models and so on
+            (implemented in inherited class).
+
+        Exception:
+            NotImplementedError . Raises exception if the method is not
+                implemented in inherited class.
+        """
         raise NotImplementedError
 
     def _assert_validation_error(self, item, error_substring):
-        """Checks that the given item passes default validation."""
+        """Checks that the given item passes default validation.
+
+        Args:
+            item: item to validate
```

Comment:
Type? Also, missing docstring for error_substring?

Reply:
not sure of type for item. Added docstring for error_substring.

Reply:
not sure of type for item. Added docstring for error_substring.